paasta_tools/secret_providers/__init__.py

```diff
@@ -49,4 +50,11 @@ def get_secret_signature_from_data(self, data: Mapping[str, Any]) -> Optional[st
 
 
 class SecretProvider(BaseSecretProvider):
-    pass
+    # from .vault import CryptoKey # cannnot import of cyclical dependency
+    def get_vault_key_versions(
+        self, key_name: str, mount_point: str = "keystore"
+    ) -> Iterable[dict]:
+        """
+        Dummy attribute to satisfy `mypy` because the class is imported dynamically via __import__
+        """
+        raise NotImplementedError
```

Comment:
should we `# type: ignore # class is imported dynamically and cannot be discovered by mypy` wherever mypy was failing rather than adding a dummy attribute?

Reply:
(i'd say we should get rid of this entire interface/indirection since it seems unlikely for us to use anything but vault for the foreseeable future, but that's probably something for future us to discuss/do :p)

Reply:
While I think this whole secretprovider thing needs to be re thought out, from basic object oriented principles, this base class shouldn't know about `vault` (or kubernetes, or aws ssm, or any other concrete implementation of SecretProvider). I would probably at least rename this to be `get_key_verisions` and any references accordingly.